lib/wallet/wallet.js

```diff
@@ -1820,6 +1825,73 @@ class Wallet extends EventEmitter {
     }
   }
 
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX with a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   * @returns {{ bid: MTX; reveal: MTX;}} {bid, reveal}
+   */
+
+  async createAuctionTxs(name, value, lockup, options) {
+    const unlock = await this.fundLock.lock();
+    try {
+      return await this._createAuctionTxs(name, value, lockup, options);
+    } finally {
+      unlock();
+    }
+  }
+
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX without a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   * @returns {{ bid: MTX; reveal: MTX;}} {bid, reveal}
+   */
+
+  async _createAuctionTxs(name, value, lockup, options) {
+    const bid = await this._createBid(name, value, lockup, options);
+
+    const bidOuputIndex = bid.outputs.findIndex(o => o.covenant.isBid());
+    const bidOutput = bid.outputs[bidOuputIndex];
+    const bidCoin = Coin.fromTX(bid, bidOuputIndex, -1);
+
+    // Prepare the data needed to make the reveal in advance
+    const nameHash = bidOutput.covenant.getHash(0);
+    let ns = await this.getNameState(nameHash);
+    if (!ns)
+      ns = await this.wdb.getNameStatus(nameHash);
+
+    const coins = [];
+    coins.push(bidCoin);
+
+    const blind = bidOutput.covenant.getHash(3);
+    const bv = await this.getBlind(blind);
+    if (!bv)
+      throw new Error('Blind value not found.');
```

Comment:
It's good you have this here, but it would be pretty crazy bug since you just made the bid (and the blind) 12 lines up!

Reply:
As discussed, left it